fastplotlib/graphics/_features/_positions_graphics.py

```diff
@@ -0,0 +1,376 @@
+        """
+        sizes = self._fix_sizes(sizes, n_datapoints)
+        super().__init__(data=sizes, isolated_buffer=isolated_buffer)
+
+    def _fix_sizes(self, sizes: int | float | np.ndarray | list[int | float] | tuple[int | float], n_datapoints: int):
+        if np.issubdtype(type(sizes), np.number):
+            # single value given
+            sizes = np.full(
+                n_datapoints, sizes, dtype=np.float32
+            )  # force it into a float to avoid weird gpu errors
+
+        elif isinstance(
+            sizes, (np.ndarray, tuple, list)
+        ):  # if it's not a ndarray already, make it one
+            sizes = np.asarray(sizes, dtype=np.float32)  # read it in as a numpy.float32
+            if (sizes.ndim != 1) or (sizes.size != n_datapoints):
+                raise ValueError(
+                    f"sequence of `sizes` must be 1 dimensional with "
+                    f"the same length as the number of datapoints"
+                )
+
+        else:
+            raise TypeError("sizes must be a single <int>, <float>, or a sequence (array, list, tuple) of int"
+                            "or float with the length equal to the number of datapoints")
+
+        if np.count_nonzero(sizes < 0) > 1:
+            raise ValueError(
+                "All sizes must be positive numbers greater than or equal to 0.0."
+            )
+
+        return sizes
+
+    def __setitem__(self, key: int | slice | np.ndarray[int | bool] | list[int | bool], value: int | float | np.ndarray | list[int | float] | tuple[int | float]):
+        # this is a very simple 1D buffer, no parsing required, directly set buffer
+        self.buffer.data[key] = value
+        self._update_range(key)
+
+        self._emit_event("sizes", key, value)
+
+
+class Thickness(GraphicFeature):
+    """line thickness"""
+    def __init__(self, value: float):
+        self._value = value
+        super().__init__()
+
+    @property
+    def value(self) -> float:
+        return self._value
+
+    def set_value(self, graphic, value: float):
+        graphic.world_object.material.thickness = value
+        self._value = value
+
+        event = FeatureEvent(type="thickness", info={"value": value})
+        self._call_event_handlers(event)
+
+
+class VertexCmap(BufferManager):
+    """
+    Sliceable colormap feature, manages a VertexColors instance and just provides a way to set colormaps.
+    """
+
+    def __init__(self, vertex_colors: VertexColors, cmap_name: str | None, cmap_values: np.ndarray | None):
+        super().__init__(data=vertex_colors.buffer)
+
+        self._vertex_colors = vertex_colors
+        self._cmap_name = cmap_name
+        self._cmap_values = cmap_values
+
+        if self._cmap_name is not None:
+            if not isinstance(self._cmap_name, str):
+                raise TypeError
+            if self._cmap_values is not None:
+                if not isinstance(self._cmap_values, np.ndarray):
+                    raise TypeError
```

Comment:
In a previous example with lines we had setting the `cmap_values` from a list, with this change we would no longer be supporting

for now, I changed the example to just cast the list to an array